src/plugin/memory-webview-main.ts

```diff
@@ -200,18 +185,12 @@ export class MemoryWebview implements vscode.CustomReadonlyEditorProvider {
         `;
     }
 
-    protected setWebviewMessageListener(panel: vscode.WebviewPanel, options?: MemoryOptions): void {
+    protected setWebviewMessageListener(panel: vscode.WebviewPanel, initialOptions?: MemoryOptions): void {
         const participant = this.messenger.registerWebviewPanel(panel);
-
+        let memoryOptions = initialOptions;
         const disposables = [
-            this.messenger.onNotification(readyType, () => {
-                this.setInitialSettings(participant, panel.title);
-                this.setSessionContext(participant, this.memoryProvider.createContext());
-                this.refresh(participant, options);
-            }, { sender: participant }),
-            this.messenger.onRequest(setOptionsType, o => {
-                options = { ...options, ...o };
-            }, { sender: participant }),
+            this.messenger.onNotification(readyType, () => this.initialize(participant, panel, memoryOptions), { sender: participant }),
+            this.messenger.onRequest(setOptionsType, newOptions => { memoryOptions = { ...memoryOptions, ...newOptions }; }, { sender: participant }),
```

Comment:
I'm not sure I understand what's going on here. We reassign the `memoryOptions` when a `setOptions` message is requested, but we only use it when we call `this.initialize` in response to the `readyType` notification, which should only happen once and should happen before the view sends a `setOptions` message. Under what circumstances would we used the reassigned value of `memoryOptions`?

Reply:
Hmm... it's been too long and I don't really remember why I did that. May have had something to do with the view state changes that may happen earlier? Not sure but I can no longer produce any issues with the previous code, so I restored it.